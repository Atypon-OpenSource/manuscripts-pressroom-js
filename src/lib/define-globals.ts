import { JSDOM } from 'jsdom'

export const defineGlobals = (): void => {
  const { window } = new JSDOM()

  Object.defineProperties(global, {
    window: {
      value: window,
    },
    document: {
      value: window.document,
    },
    DOMParser: {
      value: window.DOMParser,
    },
    Element: {
      value: window.Element,
    },
    XMLSerializer: {
      value: window.XMLSerializer,
    },
    XPathResult: {
      value: window.XPathResult,
    },
  })
}
