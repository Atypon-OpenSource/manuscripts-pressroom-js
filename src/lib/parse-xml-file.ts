import fs from 'fs-extra'

export const parseXMLFile = async (path: string): Promise<Document> => {
  const xml = await fs.readFile(path, 'UTF-8')

  return new DOMParser().parseFromString(xml, 'application/xml')
}
