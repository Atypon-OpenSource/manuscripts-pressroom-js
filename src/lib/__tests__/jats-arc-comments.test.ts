/*!
 * Â© 2020 Atypon Systems LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import {
  ContainedModel,
  HighlightableModel,
  isHighlightableModel,
  parseJATSArticle,
} from '@manuscripts/manuscript-transform'
import fs from 'fs-extra'
import randomstring from 'randomstring'

import { defineGlobals } from '../define-globals'
import {
  markCommentsWithTokens,
  replaceTokensWithHighlights,
} from '../jats-arc-comments'
import { parseXMLFile } from '../parse-xml-file'

defineGlobals()
describe('eXtyles JATS comments', () => {
  const mockInputFile =
    __dirname + '/../__mocks__/__fixtures__/index.manuscript-json'
  test('Comments highlight with random positions', async () => {
    const { data }: { data: Array<ContainedModel> } = fs.readJSONSync(
      mockInputFile
    )
    const map = new Map<string, string>()
    const changeContent = (content: string) => {
      const position = Math.floor(Math.random() * (content.length + 1))
      const token = randomstring.generate(7)
      map.set(token, 'test')
      return content.substr(0, position) + token + content.substr(position)
    }
    const modelMap = new Map(data.map((model) => [model._id, model]))
    const markedData = data.slice(0)
    for (const model of markedData) {
      if (isHighlightableModel(model)) {
        const { contents, title, caption } = model
        if (contents) {
          model.contents = changeContent(contents)
        }
        if (title) {
          model.title = changeContent(title)
        }
        if (caption) {
          model.caption = changeContent(caption)
        }
      }
    }

    replaceTokensWithHighlights(map, markedData)

    for (const model of markedData) {
      if (isHighlightableModel(model)) {
        const { contents, title, caption, _id } = model
        const originalModel = modelMap.get(_id)
        if (originalModel && isHighlightableModel(originalModel)) {
          const {
            contents: originalContent,
            title: originalTitle,
            caption: originalCaption,
          } = originalModel as HighlightableModel
          expect(contents).toStrictEqual(originalContent)
          expect(title).toStrictEqual(originalTitle)
          expect(caption).toStrictEqual(originalCaption)
        }
      }
    }
  })
  const jatsArcInput =
    __dirname + '/../__mocks__/__fixtures__/manuscript-jats-arc.xml'
  test('Comments highlight against snapshot', async () => {
    const doc = await parseXMLFile(jatsArcInput)
    const authorQueriesMap = markCommentsWithTokens(doc)
    const manuscriptModels = parseJATSArticle(doc) as ContainedModel[]
    replaceTokensWithHighlights(authorQueriesMap, manuscriptModels)
    const removeTags = (content: string) => {
      const node = new DOMParser().parseFromString(
        `<div>${content}</div>`,
        'application/xml'
      ).firstChild
      if (node) {
        return node.textContent
      }
    }
    manuscriptModels.forEach((model) => {
      if (isHighlightableModel(model)) {
        const { contents, title, caption } = model
        if (contents) {
          expect(removeTags(contents)).toMatchSnapshot()
        }
        if (title) {
          expect(removeTags(title)).toMatchSnapshot()
        }
        if (caption) {
          expect(removeTags(caption)).toMatchSnapshot()
        }
      }
    })
  })
})
