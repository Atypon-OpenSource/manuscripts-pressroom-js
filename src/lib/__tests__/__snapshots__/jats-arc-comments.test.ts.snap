// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eXtyles JATS comments Comments highlight against snapshot 1`] = `"Impact of Coronavirus on Higher Education in Pakistan: Closure of Universities, Online Teaching, Policies, Difficulties and Suggestions"`;

exports[`eXtyles JATS comments Comments highlight against snapshot 2`] = `"An Unprecedented Global Crisis! The Global, Regional, National, Political, Economic and Commercial Impact of the Coronavirus Pandemic."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 3`] = `"Effects of Coronavirus on the World Community."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 4`] = `"Closure of Universities Due to Coronavirus Disease 2019 (COVID-19): Impact on Education and Mental Health of Students and Academic Staff."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 5`] = `"Abstract"`;

exports[`eXtyles JATS comments Comments highlight against snapshot 6`] = `"Originated from Wuhan city, China, the novel coronavirus has swiftly spread throughout the world proving to be globally problematic. A complete lockdown throughout the globe has drastically effected the social life, world’s economy has badly suffered impacting education and tourism along with health issues. Fatality and continuity of COVID-19 have culminated in the closure of the world’s business. A large number of universities across the globe have suspended their academic activities, and universities are closed. Hence, online teaching is adopted by universities, globally that seems to create many issues both for the faculty and students, especially in underdeveloped countries. Similar difficulties are being faced by academia in Pakistan. This revelatory article is written in order to analyze the impact of COVID-19 on higher education in Pakistan due to the closure of universities. Under the schema of education, this paper has analyzed the prevalent situation that is concluded by recommendations and suggestions."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 7`] = `"Introduction"`;

exports[`eXtyles JATS comments Comments highlight against snapshot 8`] = `"Education is a process that is directly related to the development of a country especially for the third world developing countries it seems to be a panacea if not for every problem yet, at least most of the issues are handled with proper education. It has been observed that education is a backbone of country like ours for educational progress leads to the economic development as well. Particularly higher education plays a significant role in Pakistan’s socio-economic and political progress. Unfortunately, the current coronavirus pandemic has almost halted the educational flow throughout the world, Pakistan being no exception in facing the same issues. Therefore, the western and developed countries have adopted the online teaching methodology commonly known as distance learning or virtual learning. The evidence suggests that countries like USA, UK, Australia, France, UAE and Canada etc. are more or less already equipped with the online teaching facilities and conditions along with virtually trained teachers/students, and highly qualified IT experts. Thereof, it might not be as complicated for these countries to teach online as the third world countries like ours. Though the above mentioned countries are also facing issues related to online teaching still they are doing far much better than us."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 9`] = `"Following the steps of developed countries Pakistan has also opted for the online university teaching during the lockdown that does not seems to be efficacious due to many factors such as; remote locations of the students (most of them live in the periphery where they don’t even have proper access to electricity and prompt internet supply), teachers are not well trained for online teaching and a proper online teaching module/ syllabus is not yet available (apart from some universities who have designed a few courses yet that is not succinct), and lack of IT facilitated and equipped labs. Due to such circumstances it is conceivably become difficult for the universities to cope with online teaching that should fulfill the six requirements of distance learning as given on the Pakistan’s HEC (2020) official website:"`;

exports[`eXtyles JATS comments Comments highlight against snapshot 10`] = `"Vice Chancellors have been requested to proceed with online classes if and only if they can certify personally that six key elements are \\"online ready\\": (a) University Readiness, i.e., an effective and operational learning management system (LMS) as well as an oversight body responsible for certifying courses as online ready; (b) Faculty Readiness, i.e., faculty members have gone through training in online teaching before allowed to teach such a course; (c) Course Readiness i.e., all key information about a course is available on the LMS; (d) Library Readiness i.e., all course readings and assignments are available through online means; (e) Technology Readiness i.e., the technology needed for delivering online classes is ready for deployment; and (f) Student Readiness i.e., students are assisted in overcoming any obstacles they may have in accessing the classes and materials."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 11`] = `"It rarely seems possible for all the universities throughout Pakistan to accomplish the above mentioned prerequisites of online teaching specifically for the less assisted universities such as of Balochistan and interior Sindh. Even if the university management and IT administrations tries their level best then also the prompt internet accessibility and timely availability of electricity in the remote areas for students is more than impossible. How is it possible to teach online with access to each and every university student that may live in any periphery area of the province? Does such an online teaching will not sacrifice the quality of higher education? Unless the main purpose of online teaching is merely to complete the syllabus and compete the semester deadlines, such kind of teaching probably cannot produce the essence of learning, which face to face intimacy creates producing intellectual debates, encouraging healthy discussions that gives way to paralogy that is true scholarship rather than completion of online robotic courses. Are Pakistani students so much dedicated and virtually trained to meet the world challenges of higher education whether online or else? Are Pakistani university teachers so much competent and virtually equipped to meet the international challenges and transform the students to intellectually rich scholars and researchers? What about the students who do not even have a laptop or internet facility? How is it probable to teach music, theatre, art, medical, engineering and other practical courses online? What will be the strategies for the students whose courses cannot be taught virtually through distance learning? These are some of the queries that take place in the minds of rational persons in terms of online teaching."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 12`] = `"Certainly, we are still lacking far behind in the arena of research when compared to the international levels; online teaching will further deteriorate the higher education and will totally deplete it from research and scholarship. Therefore, it seems that blindly following the western and developed countries is not the solution to our problems, with the rise of coronavirus pandemic. Hence, considering the technological conditions of universities, already prevalent educational situations throughout the country, virtually untrained students and teachers, and the resources of the country the need is to develop our own higher education policies and solutions instead of ensuing the lead of highly developed countries in case of education."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 13`] = `"This study is an interpretative effort that has analyzed the impact of coronavirus on higher education and it is limited to Pakistan alone. Apart from analysis this study also infers some suggestions and solutions for the educational policy makers in order to improve the situation related to university education. Based on its interpretative nature the study aims to provide useful recommendations to the educationalists during the grave risk of COVID-19. It is an endeavor to minimize the deterioration of higher education through the recommendations and suggestions provided by the researchers that is limited to the interpretative analysis. This study is significant for its outline of analysis based on education as a key component of development that is impacted by the coronavirus breakout. The literature review projects that likely up till now no such study is conducted specifically in Pakistan and generally throughout the globe that is thoroughly based on the devastating impact of corona on higher education. Though different researches are conducted on the impact of coronavirus on tourism, students’ mental health due to the closure of universities and economy within the paradigm of social sciences."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 14`] = `"Background"`;

exports[`eXtyles JATS comments Comments highlight against snapshot 15`] = `"It ought to be essential to know the background of coronavirus and its emergence in order to analyze its impact on any social phenomenon. Coronavirus according to the World Health Organization belongs to a “… family of viruses that cause illnesses ranging from the common cold to more severe diseases such as Severe Acute Respiratory Syndrome (SARS) and the Middle East Respiratory Syndrome (MERS)” (Abodunrin, Oloye, and Adesola 2020, 14). Coronaviruses are the type of viruses that are spread through animals among the humans i.e. SARS is transmitted from the civet cats to humans whereas MERS is conveyed to humans from a species of camel. The historical basis of the terms coronavirus suggests that is comes from the Latin word ‘corona’, that means a crown of halo. Adodunrin, Oloye, and Adesola (2020) define that the novel coronavirus was recognized by the Chinese authorities on January 7, 2020 and they have linked it to SARS-CoV-2."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 16`] = `"According to WHO (2020) it is a fresh strain that was not earlier diagnosed in humans. The novel coronavirus is a kind of mysterious viral flow about which very little is known although human to human transmission is confirmed yet up till now there is no proper medical treatment and vaccine for the cure of this virus. That is why it has become difficult to control the spread of virus, hence it has become a fatal disease though there are many recoveries still the fatality caused by coronavirus is not a joke. The number of infected people are raising day by day and hourly deaths are being globally reported since the detection of coronavirus in Wuhan, China by the end December, 2019. For these reasons WHO (2020) has declared the coronavirus epidemic as a pandemic. Despite the precautionary measure taken by the Chines government and locking down the Wuhan city on 23rd January, 2020 COVID-19 extended to the other counties and consequently throughout the world. According to the data provided by WHO till April 8, coronavirus has spread to more than 177 countries while it has infected 3,118, 871 people throughout the world that has resulted in approximately, 216, 221 deaths."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 17`] = `"Sahu (2020) has asserted, in order to reduce the risk of COVI-19 many countries including China have imposed travel restrictions."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 18`] = `"Public health experts and government officials are taking several measures, including social distancing, self-isolation, or quarantine; strengthening health facilities to control the disease; and asking people to work at home. Several countries have announced the closure of gyms, museums, movie theaters, swimming pools, and places with large gatherings, inclusive of educational institutions, to fight this invisible enemy. (1)"`;

exports[`eXtyles JATS comments Comments highlight against snapshot 19`] = `"Furthermore, in order to decrease the spread of coronavirus among younger and adult individuals there is a global closure of school, colleges, universities and many other educational/none-educational institutions."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 20`] = `"Literature Review"`;

exports[`eXtyles JATS comments Comments highlight against snapshot 21`] = `"The following literature review incorporates a description of researches and studies (social sciences) conducted on the outbreak of coronavirus and its impact on social life till April 2020. This review is a collection of research articles that deals with the effects of coronavirus on different socio-economic spheres. Abodunrin, Oloye, and Adesola (2020) have conducted a research on the implication of coronavirus on global economy. They assert that COVID-19 is a pandemic that is not only restricted to health issues rather its consequences are vast and varied in terms of socio-economic paradigms. The virus recognizes no boundaries, thus it has cut across all the societal echelons as such; “… social, economic, education, political and human security” (13). Abodunrin, Oloye, and Adesola (2020) have critically analyzed that how COVID 19 has affected the economy throughout the globe, furthermore they have projected some methods that can curb the spread of the pandemic. Their study revealed that the pandemic has significantly caused a grim holdup to the global economic growth and development."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 22`] = `"Fetzer, Hensel, Hermle, and Roth (2020) in their research have focused upon the economic anxiety raised due to the corona pandemic. They have analyzed the global binge of coronavirus that has effected existing economy throughout the world. They have searched the global internet data that indicates the economic anxieties. They have discovered that due to the spread of coronavirus there is a speedy heave of economic anxieties in the US and other developed countries of the world. Their research recommends public education related to COVID-19 for effective control and a need for opportune procedures that can decrease economic destitution during such a central inclusive pandemic."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 23`] = `"Sahu (2020) has conducted a research on the influence of COVD-19 on education related to the mental health of the students and academic staff. He suggests that the closure of educational institutions has impacted more than 80% of the world’s students. A large number of universities throughout the world have cancelled conferences, seminars, workshops and sports while some have opted for online/virtual conferences. The universities are now in a transition phase from face-to-face interaction to online teaching methodology. The probable impact of COVID-19 on education deals with the mental health of the students as shifting from face-to-face to online classes requires a great deal of virtual training both for the students and teachers. Some of the faculty are not techno-savvy same goes for the students under such circumstances online classes tend to become a headache for them instead of facilitating the students. Besides it is also a grave psychological issue for the students who have gone to their homes in hope of returning back to their particular institutions."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 24`] = `"Igwe (2020) has assessed coronavirus as an economic doom with imminent global health issues. His article examines the repercussions to global economy, education, social activities and health. It creates a deeper understanding of the pandemic challenging all the probabilities and calculations. Apart from health and economy Igwe (2020) has also focused the issues of international students seeking higher education in countries like USA, UK, Australia, Germany and Canada. This trend of studying as an international student is currently under threat due to the coronavirus outbreak and closing of all the international borders that also includes cross-cultural exchange of students. Evans (2020) has analyzed the socio-economic impacts of coronavirus along with some solutions and policies. He has opted for proactive management approach that includes; health policy agenda, national and international shifts in investments, public and private partnership with the establishment of technical council on coronavirus."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 25`] = `"Karabag (2020) explains that the eruption of COVID-19 has produced a global, national, regional, political, economic and commercial crisis. Mahmud (2020) has specifically discussed the after effects of coronavirus on global economy that has become a most frequented angle of social scientists. Bouey (2020) has discussed the implications of COVID-19 on China’s economy related to small and medium sized enterprises. Whereas, Ying, Wang, Liu, Wen and Goh (2020) have analyzed the effects of coronavirus on tourism in China. Khan, Hassan, Fahad and Naushad (2020) have assessed the sway of COVID-19 on tourism industry and global economy. They suggest that as tourism industry plays a great rule in world’s economy therefore the influx of coronavirus has badly provoked tourism and consequently the global economy has also suffered. Khan and Naushad (2020) have discussed the effects of COVID-19 on world community that has produced far reaching concerns for the entire globe as it has effected the contemporary lifestyle and halted almost every kind of social activities and interactions."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 26`] = `"A comprehensive analysis of the available literature has enabled the researchers to detect and ascertain the gaps in the prevalent and current researches available on COVID-19 within the expanse of social sciences. Under the light of these gaps researchers have developed the research questions."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 27`] = `"Discussion and Conclusion"`;

exports[`eXtyles JATS comments Comments highlight against snapshot 28`] = `"As discussed above the impact of coronavirus is not restricted to health alone rather undoubtedly, this pandemic has impacted the very social setup. To some extent the social activities have almost become static due to the spread of virus. Certainly, education has been obstructed in many ways due to COVID-19. There are deferrals in systematized testing, evaluation and assessment systems of the educational institutions throughout the world. The evidence suggests that many institutions are even considering to extend or adjourn the teaching year due to such kind of interruptions and hindrances. Unlikely, online teaching seems to be quite challenging for the teachers and students in almost every country of the world. Even though most of the universities of the developed countries are technologically advance yet online teaching may have drawbacks for these countries as well that is making distance learning complicated."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 29`] = `"Generally in Pakistan the situation becomes more complex due to the less virtually trained faculty and students. The entire universities of Pakistan may not be technologically apt for online teaching there are lack of infrastructure that includes both the software and hardware technologies. IT department of some of the universities does not work in order to meet the current situation by providing virtual access to each and every student. There are no proper online teaching modules required for online teaching for the entire courses/syllabus need to get upgraded as per the needs of virtual classes. Not to forget that some of the degree programs cannot be teach online, there are many courses that are not possible to teach online. It has been indicated that most of the university students in Pakistan are of average level at times they are not even able to concentrate and keep pace with the lecture while sitting in the class therefore, possibilities are that they may show lack of concentration."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 30`] = `"Considering the above discussion it is not wrong to conclude that definitely, teaching online is a serious and dire issue that will affect the quality of education therefore, it needs prompt and appropriate attention by the educationalists and policy makers. This study has emphasized upon some thought provoking queries and questions in the above given introduction. These questions require some solutions that are recommended and suggested below by the researchers. Although we do not promise a complete eradication of the mentioned queries and issues related to higher education still these recommendations might may provide timely and beneficial remedies for the problem."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 31`] = `"Recommendations and Suggestions"`;

exports[`eXtyles JATS comments Comments highlight against snapshot 32`] = `"This study has brought forward the following recommendations and suggestions considering the health and other social issues:It appears that palliative and mollifying processes should be adopted by the government instead of a strict lockdown and haphazardly following the trends of developed countries. To some degree the higher education and policy makers should allow healthy administrative and academic officials to join the universities so that they can help in running the universities along with discussions and implementations for proper virtual setup, complete online based syllabus and curriculum. These measures are not only for the universities rather as suggested the entire governmental body can work by adopting the suggested measures. Here the focus of analysis is higher education thereof, the measurements are recommended for the universities. There are chances that due such measures somehow, the socio-economic agencies will also suffer less. It is important to note the above suggestion is only valid when there is a dire need to conduct online classes and we have lost all the faith in opening the education sector. Instead of conducted unplanned, untrained and shallow online teaching the best suitable step right now seems to wait for the decrease in pandemic before implementing any fruitless measures and policies. As it is indicated after a couple of months there might be a considerable amount of reduction in COVID-19 subsequently, the universities will get materially operational, incase if the virus prevails we can always adopt the recommendation number one. Meanwhile university faculty should be given proper virtual trainings so that the chances are they may become technologically equipped. Probably, it will not be an easy task for the university teachers as apart from teaching their role will have to keep on shifting between a teacher and mentor in order provide psychological and educational facilitations. It is suggested that students and faculty should receive regular emails from the university management and administration for suitable and appropriate directions. Possibly, the Higher education commission should help the less facilitated IT department of the universities to purchase and install proper virtual software and infrastructures. The IT officials should be given proper trainings for the mentioned purpose while greater level of effort is also required from the departments of Quality Assurance, ORIC and Research Directorate."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 33`] = `"References"`;

exports[`eXtyles JATS comments Comments highlight against snapshot 34`] = `"The in-text citation \\"HEC (2020)\\" is not in the reference list. Please correct the citation, add the reference to the list, or delete the citation."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 35`] = `"The in-text citation \\"Abodunrin, Oloye, and Adesola 2020\\" is not in the reference list. Please correct the citation, add the reference to the list, or delete the citation."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 36`] = `"The in-text citation \\"Adodunrin, Oloye, and Adesola (2020)\\" is not in the reference list. Please correct the citation, add the reference to the list, or delete the citation."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 37`] = `"The citation \\"WHO (2020)\\" matches multiple references. Please add letters (e.g. \\"Smith 2000a\\"), or additional authors to the citation, to uniquely match references and citations."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 38`] = `"The citation \\"WHO (2020)\\" matches multiple references. Please add letters (e.g. \\"Smith 2000a\\"), or additional authors to the citation, to uniquely match references and citations."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 39`] = `"The in-text citation \\"Abodunrin, Oloye, and Adesola (2020)\\" is not in the reference list. Please correct the citation, add the reference to the list, or delete the citation."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 40`] = `"The in-text citation \\"Abodunrin, Oloye, and Adesola (2020)\\" is not in the reference list. Please correct the citation, add the reference to the list, or delete the citation."`;

exports[`eXtyles JATS comments Comments highlight against snapshot 41`] = `"The in-text citation \\"Khan, Hassan, Fahad and Naushad (2020)\\" is not in the reference list. Please correct the citation, add the reference to the list, or delete the citation."`;
